### PR TITLE
[OSDEV-990] Create a ContriCleaner facade class for interacting with the client code.

### PR DESCRIPTION
**Pull request by vladsha-dev**
[[OSDEV-990](https://opensupplyhub.atlassian.net/browse/OSDEV-990)] Create a ContriCleaner facade class for interacting with the client code.

Implement a ContriCleaner facade class to simplify interaction with client code. With this change, the client code only needs to instantiate the ContriCleaner class, pass the input data, and then call the `process_data` method without the need to define strategies or other details. This abstraction helps streamline the process and encapsulate complexity.

### CODE REVIEW

**Review by killev — CHANGES_REQUESTED** [review]
Vlad, could you please split this PR into several smaller so that I see which refactoring you do on every step?

**Comment by vladsha-dev** [comment]
> Vlad, could you please split this PR into several smaller so that I see which refactoring you do on every step?

No, I can't. I have already done it as small as possible.

**Review by roman-stolar — APPROVED** [review]
In general LGTM.

File: src/django/contricleaner/tests/sector_cache_mock.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
-from contricleaner.lib.sector_cache_interface import SectorCacheInterface
+from contricleaner.lib.client_abstractions.sector_cache_interface import (
+    SectorCacheInterface
+)
 
 
 class MockSectorCache(SectorCacheInterface):
```
Comment (mazursasha1990) [inline src/django/contricleaner/tests/sector_cache_mock.py:6]: I suppose it's suitable to rename the class for more explicit after renaming the file name.

File: src/django/api/views/facility/facilities_view_set.py [hunk 2]
```diff
@@ -583,11 +581,9 @@ def create(self, request):
 
         log.info(f'[API Upload] Uploading data: {request.data}')
         log.info('[API Upload] Started CC Parse process!')
-        split_pattern = r', |,|\|'
```
[the remark this answers is not in the document]
Comment (vladsha-dev) [reply src/django/api/views/facility/facilities_view_set.py:586]: I don't believe the client code (Django) should handle the `split_pattern`, as it primarily pertains to ContriCleaner's functionality. I don't see it as a value that changes dynamically based on user input or other factors. Therefore, let's move it to the ContriCleaner.

File: src/django/contricleaner/tests/sector_cache_mock.py (hunk 1 above)
Comment (vladsha-dev) [reply src/django/contricleaner/tests/sector_cache_mock.py:6]: Will do.
Thanks👍

Comment (vladsha-dev) [reply src/django/contricleaner/tests/sector_cache_mock.py]: Done

**Review by mazursasha1990 — APPROVED** [review]
LGTM

**Pull request merged**